src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -450,23 +458,52 @@ void StorageRabbitMQ::bindQueue(size_t queue_id)
 
     AMQP::Table queue_settings;
 
-    queue_settings["x-max-length"] = queue_size;
+    /// Check user-defined settings.
+    if (!queue_settings_list.empty())
+    {
+        for (const auto & setting : queue_settings_list)
+        {
+            Strings setting_values;
+            splitInto<'='>(setting_values, setting);
+            assert(setting_values.size() == 2);
+            String key = setting_values[0], value = setting_values[1];
+
+            std::unordered_set<String> integer_settings = {"x-max-length", "x-max-length-bytes", "x-message-ttl", "x-expires", "x-priority", "x-max-priority"};
+            std::unordered_set<String> string_settings = {"x-overflow", "x-dead-letter-exchange", "x-queue-type"};
```

Comment:
Do we really need to create these sets every iteration in a loop?

Reply:
oh, yes, thank you.